dsp_interview_transcripts/pipeline/prep_output_tables.py

```diff
@@ -0,0 +1,160 @@
+            along with fields specified in `color` and `tooltip`.
+        color (str, optional): Encoding specification for the color channel. Defaults to "Name:N".
+        tooltip (List[str], optional): List of fields to display as tooltips. Defaults to ["Name:N", "Description:N", "question:N", "text_clean:N"].
+        domain (Optional[List[Union[str, int]]], optional): Custom domain values for the color scale, defining specific categories.
+            Defaults to None.
+        range_ (Optional[List[str]], optional): Custom color range for the color scale, corresponding to the domain values.
+            Defaults to None.
+
+    Returns:
+        alt.Chart: An Altair chart object representing the scatterplot, with specified color, tooltips, and interactivity.
+    """
+
+    if domain is not None and range_ is not None:
+        color = alt.Color(color, scale=alt.Scale(domain=domain, range=range_))
+    else:
+        color = alt.Color(color)
+
+    fig = (
+        alt.Chart(data_viz)
+        .mark_circle(size=50)
+        .encode(
+            x=alt.X(
+                "x:Q",
+                axis=alt.Axis(ticks=False, labels=False, title=None, grid=False),
+            ),
+            y=alt.Y(
+                "y:Q",
+                axis=alt.Axis(ticks=False, labels=False, title=None, grid=False),
+            ),
+            color=color,
+            opacity=opacity_condition,  # Ensure opacity_condition is defined elsewhere
+            tooltip=tooltip,
+        )
+        .properties(width=900, height=600)
+        .interactive()
+    )
+
+    return fig
+
+
+if __name__ == "__main__":
+    rep_docs = pd.read_csv(PROJECT_DIR / "outputs/user_messages_min_len_9_w_sentiment_topics_representative_docs.csv")
+    data = pd.read_csv(PROJECT_DIR / "outputs/user_messages_min_len_9_w_sentiment_topics.csv")
+    data_w_names = pd.read_csv(
+        PROJECT_DIR / "outputs/user_messages_min_len_9_w_sentiment_topics_with_names_descriptions.csv"
+    )
+
+    topic_counts = pd.DataFrame(data["Cluster"].value_counts()).reset_index()
+    topic_counts = topic_counts.rename(columns={"count": "N responses in topic"})
+
+    data_w_names = data_w_names.rename(columns={"llama3.2_name": "Name", "llama3.2_description": "Description"})
+    data_w_names = pd.merge(data_w_names, topic_counts, left_on="Cluster", right_on="Cluster", how="left")
+
+    data_w_names[["Name", "Description", "Top Words", "N responses in topic"]].to_csv(OUTPUT_PATH_SUMMARY, index=False)
+
+    rep_docs = pd.merge(
+        rep_docs, data_w_names[["Cluster", "Name", "Description", "N responses in topic"]], on="Cluster", how="left"
+    )
+
+    data_viz = pd.merge(data, data_w_names[["Cluster", "Name", "Description"]], on="Cluster", how="left")
+
+    data_viz["Name"] = data_viz["Name"].fillna("None")
+    data_viz["Description"] = data_viz["Description"].fillna("None")
```

Comment:
Also can use chains here and, for example, the function `.assign(Name = lambda df: df['Name'].fillna("None"))`